lib/astutils.cpp

```diff
@@ -3283,6 +3295,11 @@ ExprUsage getExprUsage(const Token* tok, int indirect, const Settings* settings,
             !parent->isUnaryOp("&") &&
             !(astIsRHS(tok) && isLikelyStreamRead(cpp, parent)))
             return ExprUsage::Used;
+        if (isLeafDot(tok)) {
+            const Token* top = parent->astTop();
+            if (Token::Match(top, "%assign%|++|--") && top->str() != "=")
```

Comment:
I am not sure if you should look at the "top". I assume no warning is shown in the test cases if the `++s.i;` code is changed to:

    if (++s.i < 10) {}

Reply:
You're right, fixed.